prow/pkg/githubinteractor/github.go

```diff
@@ -0,0 +1,282 @@
+package githubinteractor
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/git/v2"
+	"k8s.io/test-infra/prow/github"
+)
+
+var (
+	// Gh is used as a global object to let other packages interact with Git / GitHub
+	Gh GithubServer
+	// ErrSplit occurs, when a repository name is tried to split, but it couldn't extract the Organisation and the Repository name
+	ErrSplit = errors.New("couldn't split")
+)
+
+// InteractorOptions are the options which are used by the GitHubInteractor
+type InteractorOptions struct {
+	BotUserEmail string
+	DryRun       bool
+
+	Recursive bool
+
+	Github prowflagutil.GitHubOptions
+}
+
+// GetFileNames returns the relative filepath to the `dir`, ignoring folders / files in `ignoredFileNames`, goes through subfolders, if `recursive` is `true`
+func GetFileNames(dir string, ignoredFileNames []string, recursive bool) ([]string, error) {
+
+	files, err := os.ReadDir(dir)
+	if err != nil {
+		return []string{}, err
+	}
+
+	var fileNames []string
+	for _, file := range files {
+		for _, ignoredFileName := range ignoredFileNames {
+			if file.Name() == ignoredFileName {
+				continue
+			}
+		}
```

Comment:
doesn't do anything

either:

```suggestion
		ignored = false
		for _, ignoredFileName := range ignoredFileNames {
			if file.Name() == ignoredFileName {
				ignored = true
				break
			}
		}
		if ignored {
			continue
		}
```

or using labels

```go
abc:
for ... {
    for .. {
        // ...
        continue abc
    }
}
```